datalad/cmd.py

```diff
@@ -6,66 +6,86 @@
 #   copyright and license terms.
 #
 # ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""
+Wrapper for command calls, allowing for dry runs and output handling
+
+"""
 
-import os
-import commands
 
+import subprocess
+import sys
 import logging
-lgr = logging.getLogger('datalad.files')
+import os
+
+lgr = logging.getLogger('datalad.cmd')
 
-def dry(s, dry_):
-    """Small helper for dry runs
-    """
-    if dry_:
-        return "DRY " + s
-    return s
 
 class Runner(object):
-    """Helper to run commands and functions while allowing for dry_run and later reporting
+    """Provides a wrapper for system calls.
+
+    An object of this class provides a method to make system calls.
+    Outputs (stdout and stderr) are streamed to stdout during execution as if you were calling it from command line.
+    Additionally allows for dry runs. This is achieved by initializing the `Runner` with `dry=True`.
+    The Runner will then collect all calls as strings in `commands`.
     """
 
     __slots__ = ['commands', 'dry']
 
     def __init__(self, dry=False):
+        self.dry = dry
         self.commands = []
-        self.dry = dry                    # TODO: make use of it
 
-    # TODO -- remove dry_run specification here -- use constructor parameter
-    #def __call__(self, cmd, dry_run=False):
-    def getstatusoutput(self, cmd, dry_run=None):
-        """A wrapper around commands.getstatusoutput
+    def run(self, cmd):
+        """Runs the command `cmd` using shell.
+
+        `cmd` is called and uses stdout whereas stderr is captured and logged.
+        In case of dry-mode `cmd` is just added to `commands`.
 
-        Provides improved logging for debugging purposes and raises
-        RuntimeError exception in case of non-0 return
+        Parameters
+        ----------
+        cmd : str
+          String defining the command call.
+
+        Returns
+        -------
+        Status code as returned by the called command or `None` in case of dry-mode.
         """
-        self.log("Running: %s" % (cmd,))
+
+        self.log("Running: %s" % cmd)
```

Comment:
the reason for ```(cmd, )``` is to not blow up right here if someone passes a tuple as a cmd.  Alternative str(cmd) would move formatting outside of the string.

So if you see such a use elsewhere -- it is not a bug or useless